### PR TITLE
Add a no-op IServerAddressesFeature implementation to TestServer.Features

### LINKED ISSUE #39777: WebApplicationFactory<T>.CreateClient() fails when using app.Urls.Add() in minimal api.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

A minimal api that uses `app.Urls.Add()` throws an exception when under test with a WebApplicationFactory<T>.  The following example shows a minimal repro.

Removing the `app.Urls.Add()` works fine and the test passes.

```csharp
// Program.cs
var builder = WebApplication.CreateBuilder(args);
var app = builder.Build();

var port = Environment.GetEnvironmentVariable("PORT") ?? "8080";
app.Urls.Add($"http://0.0.0.0:{port}");

app.MapGet("/", () => "hello world");
app.Run();

public partial class Program { }
```

```csharp
// UnitTest1.cs xunit v2.4.1
[Fact]
public async Task TestAppRunning()
{
    var app = new WebApplicationFactory<Program>();
    var client = app.CreateClient(); // throws InvalidOperationException
    var response = await client.GetAsync("/");
    Assert.Equal(HttpStatusCode.OK, response.StatusCode);
}
```

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Exceptions (if any)

```text
[xUnit.net 00:00:01.02]     test.UnitTest1.TestAppRunning [FAIL]
  Failed test.UnitTest1.TestAppRunning [94 ms]
  Error Message:
   System.InvalidOperationException : IServerAddressesFeature could not be found.
  Stack Trace:
     at Microsoft.AspNetCore.Builder.WebApplication.get_Urls()
   at Program.<Main>$(String[] args) in Program.cs:line 5
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Testing.DeferredHostBuilder.DeferredHost.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.Start(IHost host)
   at Microsoft.AspNetCore.Mvc.Testing.WebApplicationFactory`1.CreateHost(IHostBuilder builder)
   at Microsoft.AspNetCore.Mvc.Testing.WebApplicationFactory`1.ConfigureHostBuilder(IHostBuilder hostBuilder)
   at Microsoft.AspNetCore.Mvc.Testing.WebApplicationFactory`1.EnsureServer()
   at Microsoft.AspNetCore.Mvc.Testing.WebApplicationFactory`1.CreateDefaultClient(DelegatingHandler[] handlers)
   at Microsoft.AspNetCore.Mvc.Testing.WebApplicationFactory`1.CreateDefaultClient(Uri baseAddress, DelegatingHandler[] handlers)
   at Microsoft.AspNetCore.Mvc.Testing.WebApplicationFactory`1.CreateClient(WebApplicationFactoryClientOptions options)
   at Microsoft.AspNetCore.Mvc.Testing.WebApplicationFactory`1.CreateClient()
   at test.UnitTest1.TestAppRunning() in UnitTest1.cs:line 23
--- End of stack trace from previous location ---
```

### .NET Version

6.0.101

### Anything else?

.NET SDK (reflecting any global.json):
 Version:   6.0.101
 Commit:    ef49f6213a

Runtime Environment:
 OS Name:     debian
 OS Platform: Linux
 RID:         linux-x64

Host (useful for support):
  Version: 6.0.1
  Commit:  3a25a7f1cc

.NET SDKs installed:
  6.0.101

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.1
  Microsoft.NETCore.App 6.0.1

### PR DESCRIPTION
**Pull request by fvoronin**
Fixes #39777

### CODE REVIEW

File: src/Mvc/test/WebSites/SimpleWebSiteWithWebApplicationBuilder/Program.cs [hunk 1]
```diff
@@ -11,6 +11,9 @@
 
 var app = builder.Build();
 
+// just to make sure that it does not cause exceptions
+app.Urls.Add($"http://0.0.0.0:{8080}");
```
Comment (davidfowl) [inline src/Mvc/test/WebSites/SimpleWebSiteWithWebApplicationBuilder/Program.cs:15]:
```suggestion
app.Urls.Add("http://localhost:8080");
```

Comment (davidfowl) [reply src/Mvc/test/WebSites/SimpleWebSiteWithWebApplicationBuilder/Program.cs:15]: These websites also run without the test server I believe...

**Comment by mkArtakMSFT** [comment]
Thanks for your PR, @fvoronin.
The below errors are most probably unrelated to your change, so please ignore the following comment.

@dotnet/aspnet-blazor-eng I looked at the failures in the `aspnetcore-components-e2e` job and I see that all the tests have failed to start JVM with the following error:

```
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:02.18] Microsoft.AspNetCore.Components.E2ETests: 
[xUnit.net 00:00:02.18] Microsoft.AspNetCore.Components.E2ETests: > microsoft.aspnetcore.components.e2etest@0.0.1 selenium-standalone
[xUnit.net 00:00:02.18] Microsoft.AspNetCore.Components.E2ETests: > selenium-standalone "start" "--config" "/home/vsts/work/1/s/src/Shared/E2ETesting/selenium-config.json" "--javaArgs=-Dwebdriver.chrome.driver=/usr/local/share/chrome_driver/chromedriver" "--" "--host" "localhost" "--port" "39711"
[xUnit.net 00:00:02.18] Microsoft.AspNetCore.Components.E2ETests: 
[xUnit.net 00:00:02.18] Microsoft.AspNetCore.Components.E2ETests: Unrecognized option: -
[xUnit.net 00:00:02.18] Microsoft.AspNetCore.Components.E2ETests: Error: Could not create the Java Virtual Machine.
[xUnit.net 00:00:02.18] Microsoft.AspNetCore.Components.E2ETests: Error: A fatal exception has occurred. Program will exit.
[xUnit.net 00:00:02.27] Microsoft.AspNetCore.Components.E2ETests: /home/vsts/work/1/s/src/Components/test/E2ETest/node_modules/selenium-standalone/lib/check-started.js:44
[xUnit.net 00:00:02.27] Microsoft.AspNetCore.Components.E2ETests:       throw new Error(
[xUnit.net 00:00:02.27] Microsoft.AspNetCore.Components.E2ETests:             ^
[xUnit.net 00:00:02.27] Microsoft.AspNetCore.Components.E2ETests: 
[xUnit.net 00:00:02.27] Microsoft.AspNetCore.Components.E2ETests: Error: Selenium exited before it could start with code 1
```

Feels like we may be calling the `selenium-standalone` with a wrong argument. I'm not sure what it should look like, but maybe `-` character the `javaArgs` value is passed incorrectly:
> `--javaArgs=`**-**`Dwebdriver.chrome.driver=/usr/local/share/chrome_driver/chromedriver`

@javiercn do you know?

**Comment by mkArtakMSFT** [comment]
@davidfowl please do not block on the above investigation, as the failed `aspnetcore-components-e2e` check is not required. Feel free to take the PR if you think it's in good shape.

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**